Sources/Starknet/Network/StarknetRequest.swift

```diff
@@ -1,25 +1,334 @@
 public struct StarknetRequest<U: Decodable> {
     let method: JsonRpcMethod
     let params: JsonRpcParams
-    let config: HttpNetworkProvider.Configuration
-    let networkProvider: HttpNetworkProvider
-
-    public func send() async throws -> U {
-        let rpcPayload = JsonRpcPayload(method: method, params: params)
-        let response = try await networkProvider.send(
-            payload: rpcPayload,
-            config: config,
-            receive: JsonRpcResponse<U>.self
-        )
-
-        if let error = response.error {
-            throw StarknetProviderError.jsonRpcError(error.code, error.message, error.data)
-        }
-
-        guard let result = response.result else {
-            throw StarknetProviderError.unknownError
-        }
-
-        return result
+
+    public init(method: JsonRpcMethod, params: JsonRpcParams) {
+        self.method = method
+        self.params = params
+    }
+}
+
+let defaultBlockId = StarknetBlockId.tag(.pending)
+let defaultSimulationFlagsForEstimateFee: Set<StarknetSimulationFlagForEstimateFee> = []
+
+public extension StarknetRequest {
```

Comment:
The static methods don't shouldn't be in an extension of `StarknetRequest`, as this forces you to always specify the generic type of `U` in order to call the static methods, eg:

```swift
let request = StarknetRequest<String>.getSpecVersion()
```

which is very cumbersome. 

If you used a different type just to namespace these methods, you wouldn't need to specifiy the generic type and will be inferred from the return type, eg:

```swift
enum StarknetRequestMethod {
    static func getSpecVersion() -> StarknetRequest<String> {
        let params = EmptyParams()

        return StarknetRequest(method: .specVersion, params: .empty(params))
    }
}
```

and then you can use:

```swift
let request = StarknetRequestMethod.getSpecVersion()
```

and `request` will be automatically inferred to be `StarknetRequest<String>`

Please can you change this, as it will make a huge difference in the usability of the interface.


Potentially we could use a shorter namespace than `StarknetRequestMethod`. Maybe just `RequestMethod` or `StarknetMethod`.

Reply:
Or maybe `RequestBuilder` for the namespace.

Reply:
Ok, maybe `StarknetRequestBuilder`?

Reply:
Ideally it should be as short as possible as you will need to type it a lot, so I would prefer `RequestBuilder`. If a project using this library already has or uses a type called `RequestBuilder` it can always be namespaces by module name if necessary.